splice_si_api/src/main/java/com/splicemachine/si/impl/SimpleTxnFilter.java

```diff
@@ -227,25 +227,22 @@ private TxnView fetchTransaction(long txnId) throws IOException{
 
     private void addToAntiTombstoneCache(DataCell data) throws IOException{
         long txnId=data.version();
-        if(isVisible(txnId)){
-			/*
-			 * We can see this anti-tombstone, hooray!
-			 */
-            if(!tombstonedTxnRows.contains(txnId))
-                antiTombstonedTxnRows.add(txnId);
+        /*
+         * Check if we can see this anti-tombstone
+         */
+        if(antiTombstonedTxnRows.isEmpty() && !tombstonedTxnRows.contains(txnId) && isVisible(txnId)){
```

Comment:
Checking for antiTombstonedTxnRows.isEmpty() is new logic here. Effectively, we can have only one value in antiTombstonedTxnRows. Is this intended?